<a href="https://colab.research.google.com/github/sumaiyasiddiqa/Play-store-app-review-analysis/blob/main/Play_Store_App_Review_Analysis_(Sumaiya_Siddiqa).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

Google Play Store or formerly Android Market, is a digital distribution service developed and operated by Google. 
It is an official apps store that provides variety content such as apps, books, magazines, music, movies and television programs.
It serves an as platform to allow users with 'Google certified' Android operating system devices to donwload applications developed 
and published on the platform either with a charge or free of cost. With the rapidly growth of Android devices and apps,
it would be interesting to perform data analysis on the data to obtain valuable insights.

Lets look at the data which consists of two files:


**1.   playstoredata.csv**:It contains all the detials of the apps on play store

**2.   userreview.csv**  :It contains user reviews for each app.The text in each review has been pre-processed and attributed with three new features: Sentiment (Positive, Negative or Neutral), Sentiment Polarity and Sentiment Subjectivity. 








# Exploring Play Store Data





In [ ]:
#import libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
from datetime import datetime
import plotly 
plotly.offline.init_notebook_mode(connected=True)
import plotly.graph_objs as go
import warnings
warnings.filterwarnings("ignore")


In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [37]:
#Load th csv file
file_path="/content/drive/MyDrive/CAPSTONE1(play store app review analysis)/Play Store Data.csv"
ps_df=pd.read_csv(file_path)

In [40]:
#Lets randomly look at the rows of the dataset
ps_df[10:20]

App        Category  Rating  \
10                          Text on Photo - Fonteee  ART_AND_DESIGN     4.4   
11          Name Art Photo Editor - Focus n Filters  ART_AND_DESIGN     4.4   
12                   Tattoo Name On My Photo Editor  ART_AND_DESIGN     4.2   
13                            Mandala Coloring Book  ART_AND_DESIGN     4.6   
14  3D Color Pixel by Number - Sandbox Art Coloring  ART_AND_DESIGN     4.4   
15                  Learn To Draw Kawaii Characters  ART_AND_DESIGN     3.2   
16     Photo Designer - Write your name with shapes  ART_AND_DESIGN     4.7   
17                         350 Diy Room Decor Ideas  ART_AND_DESIGN     4.5   
18                    FlipaClip - Cartoon animation  ART_AND_DESIGN     4.3   
19                                     ibis Paint X  ART_AND_DESIGN     4.6   

   Reviews  Size     Installs  Type Price Content Rating        Genres  \
10   13880   28M   1,000,000+  Free     0       Everyone  Art & Design   
11    8788   12M   1,000,000+  Free     0       Everyone  Art & Design   
12   44829   20M  10,000,000+  Free     0           Teen  Art & Design   
13    4326   21M     100,000+  Free     0       Everyone  Art & Design   
14    1518   37M     100,000+  Free     0       Everyone  Art & Design   
15      55  2.7M       5,000+  Free     0       Everyone  Art & Design   
16    3632  5.5M     500,000+  Free     0       Everyone  Art & Design   
17      27   17M      10,000+  Free     0       Everyone  Art & Design   
18  194216   39M   5,000,000+  Free     0       Everyone  Art & Design   
19  224399   31M  10,000,000+  Free     0       Everyone  Art & Design   

        Last Updated Current Ver   Android Ver  
10  October 27, 2017       1.0.4    4.1 and up  
11     July 31, 2018      1.0.15    4.0 and up  
12     April 2, 2018         3.8    4.1 and up  
13     June 26, 2018       1.0.4    4.4 and up  
14    August 3, 2018       1.2.3    2.3 and up  
15      June 6, 2018         NaN    4.2 and up  
16     July 31, 2018         3.1    4.1 and up  
17  November 7, 2017         1.0    2.3 and up  
18    August 3, 2018       2.2.5  4.0.3 and up  
19     July 30, 2018       5.5.4    4.1 and up

In [ ]:
ps_df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [ ]:
#Finding the names of the columns and the number of rows and columns
print(list(ps_df.columns))
print(f'The number of rows of the dataframe playstore data are :{ps_df.shape[0]}')
print(f'The number of columns of the dataframe playstore data are: {ps_df.shape[1]}')

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
The number of rows of the dataframe playstore data are :10841
The number of columns of the dataframe playstore data are: 13


The description of columns of playstore dataframe :
   



> 

1.    App : It has the name of the application.
2.    Category : It has the category to which the app belongs.
3.    Rating :  It has the average rating given by the users in the play store.
4.    Reviews : It tells us about the total number of users who have given a review for the application.
5.    Size : It has the size of the app.
6.    Installs : It has the total no of downloads or installs.
7.    Type : It tells about the type of the app whether free or paid.
8.    Price : It has the price of the app.
9.    Content rating : It states whether or not an app is suitable for all age groups or not. 
10.   Genres : The genre of the app
11.   Last updated : It has the date on which the app was last updated.
12.   Current ver : It has the current version of the app.
13.   Android ver : It tells about the minimum android version required to run the app.










## **Data cleaning**

Removing the null values and duplicate values present in the dataset.By looking at the dataset carefully we can see that their are many special characters and letters in the columns like (Install, Price,Size) which make their conversion to numeric datatype difficult.Let's clean the data by removing all these.



In [46]:
#Check on the null values
ps_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

By above analysis we can see that 'Ratings' column has 1474 null values which is 13% of the data of ratings column, column 'Type' has 1 null value, column 'Content rating' has 1 null value, column 'Current ver' has 8 null values and column 'Android ver' has 3 null values.


Lets first deal with the columns which contains lesser number of NaN values. By going through the NaN values, we must come up with a way to replace them with non NaN values or we need to come up with a reason for having NaN.

### **1. Android ver has 3 null values**

In [45]:
#The rows containing the null values in column Android Ver
ps_df[ps_df['Android Ver'].isnull()]

App         Category  Rating  \
4453                    [substratum] Vacuum: P  PERSONALIZATION     4.4   
4490                      Pi Dark [substratum]  PERSONALIZATION     4.5   
10472  Life Made WI-Fi Touchscreen Photo Frame              1.9    19.0   

      Reviews    Size Installs  Type     Price Content Rating  \
4453      230     11M   1,000+  Paid     $1.49       Everyone   
4490      189    2.1M  10,000+  Free         0       Everyone   
10472    3.0M  1,000+     Free     0  Everyone            NaN   

                  Genres    Last Updated Current Ver Android Ver  
4453     Personalization   July 20, 2018         4.4         NaN  
4490     Personalization  March 27, 2018         1.1         NaN  
10472  February 11, 2018          1.0.19  4.0 and up         NaN

In [ ]:
#android ver has 3 null values
ps_df['Android Ver'].value_counts()

4.1 and up            2451
4.0.3 and up          1501
4.0 and up            1375
Varies with device    1362
4.4 and up             980
2.3 and up             652
5.0 and up             601
4.2 and up             394
2.3.3 and up           281
2.2 and up             244
4.3 and up             243
3.0 and up             241
2.1 and up             134
1.6 and up             116
6.0 and up              60
7.0 and up              42
3.2 and up              36
2.0 and up              32
5.1 and up              24
1.5 and up              20
4.4W and up             12
3.1 and up              10
2.0.1 and up             7
8.0 and up               6
7.1 and up               3
4.0.3 - 7.1.1            2
5.0 - 8.0                2
1.0 and up               2
7.0 - 7.1.1              1
4.1 - 7.1.1              1
5.0 - 6.0                1
2.2 - 7.1.1              1
5.0 - 7.1.1              1
Name: Android Ver, dtype: int64

Since we cannot replace the null values with any other values, lets drop the 3 rows which constitues only 0.03% of the total data of the column 

In [48]:
ps_df=ps_df[ps_df['Android Ver'].notna()]
ps_df.shape

(10838, 13)

We dropped the NaN values of Android Ver.

### **2. Current Ver has 8 Null values**

In [49]:
#The rows containing null values in the Current Ver column
ps_df[ps_df['Current Ver'].isnull()]

App             Category  Rating Reviews  \
15     Learn To Draw Kawaii Characters       ART_AND_DESIGN     3.2      55   
1553              Market Update Helper   LIBRARIES_AND_DEMO     4.1   20145   
6322            Virtual DJ Sound Mixer                TOOLS     4.2    4010   
6803                         BT Master               FAMILY     NaN       0   
7333                       Dots puzzle               FAMILY     4.0     179   
7407                   Calculate My IQ               FAMILY     NaN      44   
7730                            UFO-CQ                TOOLS     NaN       1   
10342                   La Fe de Jesus  BOOKS_AND_REFERENCE     NaN       8   

       Size    Installs  Type  Price Content Rating             Genres  \
15     2.7M      5,000+  Free      0       Everyone       Art & Design   
1553    11k  1,000,000+  Free      0       Everyone   Libraries & Demo   
6322   8.7M    500,000+  Free      0       Everyone              Tools   
6803   222k        100+  Free      0       Everyone          Education   
7333    14M     50,000+  Paid  $0.99       Everyone             Puzzle   
7407   7.2M     10,000+  Free      0       Everyone      Entertainment   
7730   237k         10+  Paid  $0.99       Everyone              Tools   
10342  658k      1,000+  Free      0       Everyone  Books & Reference   

            Last Updated Current Ver Android Ver  
15          June 6, 2018         NaN  4.2 and up  
1553   February 12, 2013         NaN  1.5 and up  
6322        May 10, 2017         NaN  4.0 and up  
6803    November 6, 2016         NaN  1.6 and up  
7333      April 18, 2018         NaN  4.0 and up  
7407       April 3, 2017         NaN  2.3 and up  
7730        July 4, 2016         NaN  2.0 and up  
10342   January 31, 2017         NaN  3.0 and up

In [50]:
#Finding the different values in the Current Ver column
ps_df['Current Ver'].value_counts()

Varies with device    1459
1.0                    809
1.1                    263
1.2                    178
2.0                    151
                      ... 
5.44.1                   1
7.16.8                   1
04.08.00                 1
2.10.06                  1
2.0.148.0                1
Name: Current Ver, Length: 2831, dtype: int64

We cannot replace the null values with any other values in the Current Ver column, So we have to drop the rows with null values

In [53]:
#Dropping the rows with null values in the Current Ver column
ps_df=ps_df[ps_df['Current Ver'].notna()]
ps_df.shape

(10830, 13)

### **3.Type column has only one null value**

In [54]:
#The row containiing the null value in the column Type
ps_df[ps_df['Type'].isnull()]

App Category  Rating Reviews                Size  \
9148  Command & Conquer: Rivals   FAMILY     NaN       0  Varies with device   

     Installs Type Price Content Rating    Genres   Last Updated  \
9148        0  NaN     0   Everyone 10+  Strategy  June 28, 2018   

             Current Ver         Android Ver  
9148  Varies with device  Varies with device

In [55]:
#Checking the values in the Type column
ps_df['Type'].value_counts()

Free    10032
Paid      797
Name: Type, dtype: int64

As the Type column has the type of the app i.e, paid or free and in the above row the price column has value 0, we can change the Type column null value to 'free'

In [56]:
#Changing the Type column null value to 'free'
ps_df.loc[9148,'Type']='Free'

In [57]:
#checking the result
ps_df[ps_df['Type'].isnull()]

Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver, Android Ver]
Index: []

We successfully replaced the null value of column Type to 'Free'

### **4.Rating column has 1474 null values**

In [58]:
#The rows containing the null values in the Rating column
ps_df[ps_df['Rating'].isnull()]


App             Category  Rating  \
23                  Mcqueen Coloring pages       ART_AND_DESIGN     NaN   
113              Wrinkles and rejuvenation               BEAUTY     NaN   
123                 Manicure - nail design               BEAUTY     NaN   
126           Skin Care and Natural Beauty               BEAUTY     NaN   
129    Secrets of beauty, youth and health               BEAUTY     NaN   
...                                    ...                  ...     ...   
10824                            Cardio-FR              MEDICAL     NaN   
10825                   Naruto & Boruto FR               SOCIAL     NaN   
10831             payermonstationnement.fr  MAPS_AND_NAVIGATION     NaN   
10835                             FR Forms             BUSINESS     NaN   
10838               Parkinson Exercices FR              MEDICAL     NaN   

      Reviews  Size  Installs  Type Price Content Rating  \
23         61  7.0M  100,000+  Free     0       Everyone   
113       182  5.7M  100,000+  Free     0   Everyone 10+   
123       119  3.7M   50,000+  Free     0       Everyone   
126       654  7.4M  100,000+  Free     0           Teen   
129        77  2.9M   10,000+  Free     0     Mature 17+   
...       ...   ...       ...   ...   ...            ...   
10824      67   82M   10,000+  Free     0       Everyone   
10825       7  7.7M      100+  Free     0           Teen   
10831      38  9.8M    5,000+  Free     0       Everyone   
10835       0  9.6M       10+  Free     0       Everyone   
10838       3  9.5M    1,000+  Free     0       Everyone   

                                Genres        Last Updated Current Ver  \
23     Art & Design;Action & Adventure       March 7, 2018       1.0.0   
113                             Beauty  September 20, 2017         8.0   
123                             Beauty       July 23, 2018         1.3   
126                             Beauty       July 17, 2018        1.15   
129                             Beauty      August 8, 2017         2.0   
...                                ...                 ...         ...   
10824                          Medical       July 31, 2018       2.2.2   
10825                           Social    February 2, 2018         1.0   
10831                Maps & Navigation       June 13, 2018   2.0.148.0   
10835                         Business  September 29, 2016       1.1.5   
10838                          Medical    January 20, 2017         1.0   

      Android Ver  
23     4.1 and up  
113    3.0 and up  
123    4.1 and up  
126    4.1 and up  
129    2.3 and up  
...           ...  
10824  4.4 and up  
10825  4.0 and up  
10831  4.0 and up  
10835  4.0 and up  
10838  2.2 and up  

[1470 rows x 13 columns]

Rating columns has 1474 null values so if we drop the rows we will loose a large amount of data.
In this case we should replace the null values with the median of the remaining values in the Rating column

In [60]:
#User defined function to find the median of the column Rating
def replace_with_median(series):
    return series.fillna(series.median())

ps_df['Rating'] = ps_df['Rating'].transform(replace_with_median)